src/controllers/productController.js

```diff
@@ -22,9 +22,48 @@ const create = async (req, res) => {
   res.redirect("/api/v1/products");
 };
 
-const showNewForm = (req, res) => {
-  res.render("products/showNewForm");
+const edit = async (req, res) => {
+  const productId = req.params.id;
+  productService
+    .findById(productId)
+    .then((prod) => {
```

Comment:
Use async await syntax as it increases code readability and we have `catchAsync` which can be utilised to to catch asynchronous error , see the existing code.. You will not need a explicit exception handling in the controller method, also we should always use global exception handler to handle any exception, this keeps code generic. Also in your catch method you are silently logging the error, we should not do that, we should throw the exception and throw it loud unless you have a special reason to handle it.

Reply:
done